### PR TITLE
fix MSVC compiler warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
and a MinGW warning too.
Don't cache (old) mingw dependencies on AppVeyor

### CODE REVIEW

File: libmscore/score.cpp [hunk 1]
```diff
@@ -4501,7 +4501,7 @@ void MasterScore::updateExpressive(MasterSynthesizer* m)
       SynthesizerState s = synthesizerState();
       SynthesizerGroup g = s.group("master");
 
-      int method;
+      int method = 0;
```
Comment (jthistle) [inline libmscore/score.cpp]: @Jojo-Schmitz - could you change this? It should really be defaulting to 1, it was a mistake on my part to make it default to nothing (i.e. 0).

Comment (Jojo-Schmitz) [reply libmscore/score.cpp]: done

Comment (jthistle) [reply libmscore/score.cpp]: Thank you!

File: libmscore/rendermidi.cpp [hunk 2]
```diff
@@ -777,7 +776,7 @@ static void collectMeasureEventsDefault(EventMap* events, Measure* m, Staff* sta
 
                         // Check if there is a fortepiano / similar dynamic
                         bool hasChangingDynamic = false;
-                        Dynamic* changingDyn;
+                        Dynamic* changingDyn = 0;
```
Comment (anatoly-os) [inline libmscore/rendermidi.cpp]: `nullptr`?

Comment (Jojo-Schmitz) [reply libmscore/rendermidi.cpp]: done

**Comment by Jojo-Schmitz** [comment]
Rebased and squashed, as apparently the 3.0.* only part hadn't been taken for 3.0.5...

**Pull request merged**